varint.go

```diff
@@ -1,44 +1,27 @@
 package multiaddr
 
 import (
-	"encoding/binary"
-	"fmt"
-	"math/bits"
-)
+	"math"
 
-// VarintSize returns the size (in bytes) of `num` encoded as a varint.
-func VarintSize(num int) int {
-	bits := bits.Len(uint(num))
-	q, r := bits/7, bits%7
-	size := q
-	if r > 0 || size == 0 {
-		size++
-	}
-	return size
-}
+	"github.com/multiformats/go-varint"
+)
 
 // CodeToVarint converts an integer to a varint-encoded []byte
 func CodeToVarint(num int) []byte {
-	buf := make([]byte, VarintSize(num))
-	n := binary.PutUvarint(buf, uint64(num))
-	return buf[:n]
+	if num < 0 || num > math.MaxInt32 {
```

Comment:
maybe we add a helper function that accepts a `uint32` and start to prefer using that directly, but have this function execute these guards and then attempt to cast and call that helper function

Reply:
Well, we _should_ have one that accepts uint64. But we don't need to tackle that yet.